first

Kovarniykrab opened the pull request.

Kovarniykrab committed:
first

Kovarniykrab commented:
Я немного накосячил когда делал pull request, потому что случайно обновил страничку и наворотил фиг пойми чего, чтобы его создать. Но вроде все гуд. Я не очень хорошо разобрался в функционалом гитхаба

Kovarniykrab commented:
Еще момент. В пул реквест довольно много изменений на первый взгляд. Дело в том что я не очень люблю заморачиваться со всякими коммитами и тд. Поэтому просто выполнял задание на основе старого такого же сервиса в среде разработки. Ну и с чувством выполненного долга сделал копи пасте и все. А так то задача не сложная. Прекод по сути только 3 проверки требует и сам запрос.
единственный момент который я так и не понял из пакета testing: 
Вот когда мы вызываем любой метод assert, почему в своих аргументах она требует первым параметром t *testing.T? а уже потом все остальное. Я документацию пока не читал

ivannizh reviewed (commented):
Привет! Тесты выполняются в отдельном файле с окончанием `_test.go`. Также нужно проверить, что тесты проходят перед отправкой командой `go test ./... -v -count=1`.

Так же по ТЗ нужно 3 теста.

Kovarniykrab commented:
А разве то что я сделал не считается тестом?

ср, 19 февр. 2025 г., 22:31 Ivan Nizhnichenko ***@***.***>:

> ***@***.**** commented on this pull request.
>
> Привет! Тесты выполняются в отдельном файле с окончанием _test.go. Также
> нужно проверить, что тесты проходят перед отправкой командой go test
> ./... -v -count=1.
>
> Так же по ТЗ нужно 3 теста.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Kovarniykrab/tistifyHomework/pull/1#pullrequestreview-2626805116>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BJVCTZZNBPZ5KJM2FQGPZP32QSBRHAVCNFSM6AAAAABXNGXNUKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMMRWHAYDKMJRGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Kovarniykrab commented:
Приветствую

ср, 19 февр. 2025 г., 22:56 Kovarniy Krab ***@***.***>:

> А разве то что я сделал не считается тестом?
>
> ср, 19 февр. 2025 г., 22:31 Ivan Nizhnichenko ***@***.***>:
>
>> ***@***.**** commented on this pull request.
>>
>> Привет! Тесты выполняются в отдельном файле с окончанием _test.go. Также
>> нужно проверить, что тесты проходят перед отправкой командой go test
>> ./... -v -count=1.
>>
>> Так же по ТЗ нужно 3 теста.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Kovarniykrab/tistifyHomework/pull/1#pullrequestreview-2626805116>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BJVCTZZNBPZ5KJM2FQGPZP32QSBRHAVCNFSM6AAAAABXNGXNUKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMMRWHAYDKMJRGY>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

ivannizh commented:
То что у тебя считается тестом, просто он 1 и не запускается через средства Го

Kovarniykrab committed:
исправленные тесты

Kovarniykrab commented:
Понял что от меня требуется. Сбил с толку прекод. Я почему-то подумал, что надо все сделать в одном файле. В общем все сделал, надеюсь сейчас все корректно.

ivannizh commented on precode_test.go:
```diff
@@ -0,0 +1,72 @@
+	// здесь нужно добавить необходимые проверки
+
+	status := responseRecorder.Code
+	assert.Equal(t, status, http.StatusBadRequest)
```
На втором месте нужно ставить что хотим увидеть, а на третьем, что проверяем.
На самом тест это не влияет, но когда он упадет, то вывод будет неправильный

ivannizh commented on precode_test.go:
```diff
@@ -0,0 +1,72 @@
+	//и я считаю, что так будет правильнее. Ведь пользователь может ввести любой запрос
+	//условно я представил, что в мапе лежит больше одного города
+	//надеюсь Вы простите меня за небольшую шалось)
+	//еще у меня почему-то тесты не запускаются, поэтому надеюсь, что все должно быть правильно
```
все норм работает

ivannizh commented on precode_test.go:
```diff
@@ -0,0 +1,72 @@
+	assert.Len(t, list, expectedLenght)
+	//я знаю, что я убрал  totalCount :=4 и было бы проще написать код такого рода:
+	// assert.Len(t, list, totalCount), но мне было очень интересно попробовать решить задачу таким образом
+	//и я считаю, что так будет правильнее. Ведь пользователь может ввести любой запрос
```
Ну так реально лучше, так что не нужно переживать

ivannizh reviewed (approved):
Отлично, принято!

Kovarniykrab commented:
Спасибо большое)

сб, 22 февр. 2025 г., 15:39 Ivan Nizhnichenko ***@***.***>:

> ***@***.**** approved this pull request.
>
> Отлично, принято!
> ------------------------------
>
> In precode_test.go
> <https://github.com/Kovarniykrab/tistifyHomework/pull/1#discussion_r1966454329>
> :
>
> > +	body := responseRecorder.Body.String()
> +	assert.NotEmpty(t, body, "тело запроса пустое")
> +
> +	assert.Equal(t, body, "Мир кофе,Сладкоежка,Кофе и завтраки,Сытый студент")
> +}
> +
> +func TestMainHandlerBadRequest(t *testing.T) {
> +	req := httptest.NewRequest("GET", "/cafe?count=1&city=Tomsk", nil) // здесь нужно создать запрос к сервису
> +
> +	responseRecorder := httptest.NewRecorder()
> +	handler := http.HandlerFunc(mainHandle)
> +	handler.ServeHTTP(responseRecorder, req)
> +	// здесь нужно добавить необходимые проверки
> +
> +	status := responseRecorder.Code
> +	assert.Equal(t, status, http.StatusBadRequest)
>
> На втором месте нужно ставить что хотим увидеть, а на третьем, что
> проверяем.
> На самом тест это не влияет, но когда он упадет, то вывод будет
> неправильный
> ------------------------------
>
> In precode_test.go
> <https://github.com/Kovarniykrab/tistifyHomework/pull/1#discussion_r1966456638>
> :
>
> > +	assert.Equal(t, status, http.StatusOK)
> +
> +	body := responseRecorder.Body.String()
> +	assert.NotEmpty(t, body, "тело запроса пустое")
> +
> +	list := strings.Split(body, ",")
> +
> +	city := req.URL.Query().Get("city")
> +	expectedLenght := len(cafeList[city])
> +	assert.Len(t, list, expectedLenght)
> +	//я знаю, что я убрал  totalCount :=4 и было бы проще написать код такого рода:
> +	// assert.Len(t, list, totalCount), но мне было очень интересно попробовать решить задачу таким образом
> +	//и я считаю, что так будет правильнее. Ведь пользователь может ввести любой запрос
> +	//условно я представил, что в мапе лежит больше одного города
> +	//надеюсь Вы простите меня за небольшую шалось)
> +	//еще у меня почему-то тесты не запускаются, поэтому надеюсь, что все должно быть правильно
>
> все норм работает
> ------------------------------
>
> In precode_test.go
> <https://github.com/Kovarniykrab/tistifyHomework/pull/1#discussion_r1966456683>
> :
>
> > +	// здесь нужно добавить необходимые проверки
> +
> +	status := responseRecorder.Code
> +	assert.Equal(t, status, http.StatusOK)
> +
> +	body := responseRecorder.Body.String()
> +	assert.NotEmpty(t, body, "тело запроса пустое")
> +
> +	list := strings.Split(body, ",")
> +
> +	city := req.URL.Query().Get("city")
> +	expectedLenght := len(cafeList[city])
> +	assert.Len(t, list, expectedLenght)
> +	//я знаю, что я убрал  totalCount :=4 и было бы проще написать код такого рода:
> +	// assert.Len(t, list, totalCount), но мне было очень интересно попробовать решить задачу таким образом
> +	//и я считаю, что так будет правильнее. Ведь пользователь может ввести любой запрос
>
> Ну так реально лучше, так что не нужно переживать
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Kovarniykrab/tistifyHomework/pull/1#pullrequestreview-2634727762>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BJVCTZ2NQINQVQFEDB6STPD2RALSHAVCNFSM6AAAAABXNGXNUKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMMZUG4ZDONZWGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
